src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,940 @@
+_CURRENCY: str = "currency"  # CCXT only variable
+_DATA: str = "data"
+_DATETIME: str = "datetime"  # CCXT only variable
+_DEPOSIT: str = "deposit" # CCXT
+_DIVTIME: str = "divTime"
+_ENDTIME: str = "endTime"
+_ENINFO: str = "enInfo"
+_FEE: str = "fee"
+_FIATCURRENCY: str = "fiatCurrency"
+_ID: str = "id"  # CCXT
+_INDICATEDAMOUNT: str = "indicatedAmount"
+_INFO: str = "info"
+_INSERTTIME: str = "insertTime"
+_ISDUST: str = "isDust"
+_ISFIATPAYMENT: str = "isFiatPayment"
+_LIMIT: str = "limit"
+_OBTAINAMOUNT: str = "obtainAmount"
+_ORDER: str = "order"  # CCXT
+_ORDERNO: str = "orderNo"
+_PAGEINDEX: str = "pageIndex"
+_PAGESIZE: str = "pageSize"
+_PRICE: str = "price"
+_PROFITAMOUNT: str = "profitAmount"
+_ROWS: str = "rows"
+_SELL: str = "sell"  # CCXT
+_SIDE: str = "side"  # CCXT
+_STARTTIME: str = "startTime"
+_STATUS: str = "status"
+_SOURCEAMOUNT: str = "sourceAmount"
+_SYMBOL: str = "symbol"
+_TIME: str = "time"
+_TIMESTAMP: str = "timestamp"  # CCXT
+_TRANID: str = "tranId"
+_TRANSACTIONTYPE: str = "transactionType"
+_TOTAL: str = "total"
+_TOTALFEE: str = "totalFee"
+_TYPE: str = "type"
+_TXID: str = "txid"  # CCXT doesn't capitalize I
+_UPDATETIME: str = "updateTime"
+_USERNAME: str = "userName"
+_WITHDRAWAL: str = "withdrawal" # CCXT
+
+# Time period constants
+_NINETY_DAYS_IN_MS: int = 7776000000
+_THIRTY_DAYS_IN_MS: int = 2592000000
+
+# Types of Binance Dividends
+_BNBVAULT = "BNB Vault"
+_ETHSTAKING = "ETH 2.0 Staking"
+_FLEXIBLESAVINGS = "Flexible Savings"
+_LOCKEDSAVINGS = "Locked Savings"
+_LOCKEDSTAKING = "Locked Staking"
+_INTEREST_LIST = [_FLEXIBLESAVINGS, _LOCKEDSAVINGS]
+_STAKING_LIST = [_ETHSTAKING, _LOCKEDSTAKING, _BNBVAULT]
+
+
+class _ProcessAccountResult(NamedTuple):
+    in_transactions: List[InTransaction]
+    out_transactions: List[OutTransaction]
+    intra_transactions: List[IntraTransaction]
+
+
+class _Trade(NamedTuple):
+    base_asset: str
+    quote_asset: str
+    base_info: str
+    quote_info: str
+
+
+class InputPlugin(AbstractInputPlugin):
+
+    __BINANCE_COM: str = "Binance.com"
+
+    def __init__(
+        self,
+        account_holder: str,
+        api_key: str,
+        api_secret: str,
+        username: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder)
+        self.__logger: logging.Logger = create_logger(f"{self.__BINANCE_COM}/{self.account_holder}")
+        self.__cache_key: str = f"{self.__BINANCE_COM.lower()}-{account_holder}"
+        self.client: ccxt.binance = ccxt.binance(
+            {
+                "apiKey": api_key,
+                "secret": api_secret,
+            }
+        )
+        self.username = username
+
+        # We have to know what markets are on Binance so that we can pull orders using the market
+        self.markets: List[str] = []
+        ccxt_markets: Any = self.client.fetch_markets()
+        for market in ccxt_markets:
+#            self.__logger.debug("Market: %s", json.dumps(market))

```

Comment:
Uncomment?

Reply:
I had to comment this out because of logger spam when testing. Is there a way to mock an attribute of an instance before it is initialized in testing?

The problem I run into is that I want to mock `client` for the InputPlugin I'm testing, but I can't figure out how to do it before I initialize it and so `client` pulls all 2000+  markets in `__init__()`. How can I mock it so it doesn't pull all the markets in the initializer?

Reply:
Oh, I see. Perhaps move the fetch_markets() and all the other network calls to load()? Basically leave the creation of the binance() client in __init__() and move the rest to load(): would that solve the problem?

Reply:
Yeah, that solves it. Thanks.